src/data.py

```diff
@@ -68,32 +67,66 @@ def dict(self) -> dict:
             for slotname in self.__slots__  # type: ignore
         }
 
-    def calculate_all(self, **kwargs):
-        """Calculate extra parameters and place them in `.extras` attribute and return self.extras."""
-        self.config = kwargs
+    def calculate_all(self,
```

Comment:
```suggestion
   from functools import cache
   @cache
    def calculate_all(self,
```
That might just work now

Reply:
Indeed, but I don't think it will give any performance benefits at this moment.